AutoInstaller should name directory based on ABI

gsnedders opened the pull request:
#### 17ed495179235a185b77e1ef4d02778d9fb1dda0
<pre>
AutoInstaller should name directory based on ABI
<a href="https://bugs.webkit.org/show_bug.cgi?id=224669">https://bugs.webkit.org/show_bug.cgi?id=224669</a>

Reviewed by Jonathan Bedard.

While the autoinstaller currently (mostly!) puts autoinstalled libraries
into a directory such as `python-3-arm64`, this doesn&apos;t suffice to
uniquely identify the ABI of the interpreter.

In practice, this mostly works because since Python 3.2 distutils will
compile extensions into files with suffixes such as
`cpython-39-darwin.so`, and most extension modules we compile are
optional &quot;speedup&quot; modules where failing to have an appropriate
extension module will simply degrade performance but not prevent
operation.

This replaces our current tags with a string that should match
`str(list(packaging.tags.sys_tags())[0])`.

It doesn&apos;t, however, use packaging directly, because we can&apos;t install
packaging until we know what directory we&apos;re using. If we do in future
move to vendoring packaging we should be able to migrate to this without
invalidating everyone&apos;s local autoinstall directories.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/webserver.py:
* Tools/Scripts/libraries/resultsdbpy/container:
* Tools/Scripts/libraries/resultsdbpy/run:
* Tools/Scripts/libraries/resultsdbpy/run-tests:
* Tools/Scripts/libraries/webkitbugspy/run-tests:
* Tools/Scripts/libraries/webkitcorepy/run-tests:
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py:
(AutoInstall._platform_compatibility_tag):
(AutoInstall):
(AutoInstall.set_directory):
* Tools/Scripts/libraries/webkitscmpy/run-tests:
* Tools/Scripts/webkitpy/__init__.py:

Canonical link: <a href="https://commits.webkit.org/268205@main">https://commits.webkit.org/268205@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/16994e0bf21d4967ba8575fef7b6a0ea8f3ee252

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16501 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16823 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17262 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18281 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15475 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16691 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20054 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16966 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17827 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16698 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17113 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14276 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19057 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/16626 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14357 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14960 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21753 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15345 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15126 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19437 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15715 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13345 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/16384 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14916 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19285 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2318 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15541 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (approved):
r+ing, but we should probably have this land early on a Monday, I this this change has a relatively high chance of causing some fallout in folks personal checkouts.

gsnedders committed:
AutoInstaller should name directory based on ABI
https://bugs.webkit.org/show_bug.cgi?id=224669

Reviewed by Jonathan Bedard.

While the autoinstaller currently (mostly!) puts autoinstalled libraries
into a directory such as `python-3-arm64`, this doesn't suffice to
uniquely identify the ABI of the interpreter.

In practice, this mostly works because since Python 3.2 distutils will
compile extensions into files with suffixes such as
`cpython-39-darwin.so`, and most extension modules we compile are
optional "speedup" modules where failing to have an appropriate
extension module will simply degrade performance but not prevent
operation.

This replaces our current tags with a string that should match
`str(list(packaging.tags.sys_tags())[0])`.

It doesn't, however, use packaging directly, because we can't install
packaging until we know what directory we're using. If we do in future
move to vendoring packaging we should be able to migrate to this without
invalidating everyone's local autoinstall directories.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/webserver.py:
* Tools/Scripts/libraries/resultsdbpy/container:
* Tools/Scripts/libraries/resultsdbpy/run:
* Tools/Scripts/libraries/resultsdbpy/run-tests:
* Tools/Scripts/libraries/webkitbugspy/run-tests:
* Tools/Scripts/libraries/webkitcorepy/run-tests:
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py:
(AutoInstall._platform_compatibility_tag):
(AutoInstall):
(AutoInstall.set_directory):
* Tools/Scripts/libraries/webkitscmpy/run-tests:
* Tools/Scripts/webkitpy/__init__.py:

Canonical link: https://commits.webkit.org/268205@main
